utils.c

```diff
@@ -296,6 +296,20 @@ int janus_mkdir(const char *dir, mode_t mode) {
 	return 0;
 }
 
+gchar* janus_make_absolute_path(const gchar *base_dir, const gchar *path)
+{
+	if(!path)
+		return NULL;
+
+	if(g_path_is_absolute(path))
+		return g_strdup(path);
+
+	if(!base_dir)
+		return NULL;
+
+	return g_build_filename(base_dir, path, NULL);
```

Comment:
For code style as well, I think you can remove all those extra empty lines to separate the checks.